realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1037,23 +1037,27 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFind(
     return -1;
 }
 
-std::unique_ptr<Query> getHandoverQuery (jlong bgSharedGroupPtr, jlong queryPtr, bool advanceToLatestVersion)
+// Returns a pointer to query on the worker SharedGroup or throw a BadVersion if the SharedGroup version required
+// for the handover is no longer available.
+std::unique_ptr<Query> handoverQueryToWorker(jlong bgSharedGroupPtr, jlong queryPtr, bool advanceToLatestVersion)
```

Comment:
I think so yes. That unit test covers the standard scenario that could cause caller and worker to get out of sync. The BadVersion in this PR is a lot more subtle as it is a race condition between async writes and queries.